src/main/java/apoc/export/graphml/ExportGraphML.java

```diff
@@ -90,14 +103,41 @@
         return exportGraphML(fileName, source, graph, c);
     }
 
-    private Stream<ProgressInfo> exportGraphML(@Name("file") String fileName, String source, SubGraph graph, ExportConfig config) throws Exception, XMLStreamException {
-        if (fileName != null) checkWriteAllowed();
+    private Stream<ProgressInfo> exportGraphML(@Name("file") String fileName, String source, SubGraph graph, ExportConfig exportConfig) throws Exception {
+        if (StringUtils.isNotBlank(fileName)) checkWriteAllowed(exportConfig);
         ProgressReporter reporter = new ProgressReporter(null, null, new ProgressInfo(fileName, source, "graphml"));
-        PrintWriter printWriter = getPrintWriter(fileName, null);
         XmlGraphMLWriter exporter = new XmlGraphMLWriter();
-        exporter.write(graph, printWriter, reporter, config);
-        printWriter.flush();
-        printWriter.close();
-        return reporter.stream();
+        ExportFileManager cypherFileManager = FileManagerFactory.createFileManager(fileName, false, exportConfig.streamStatements());
+        final PrintWriter graphMl = cypherFileManager.getPrintWriter("graphml");
+        if (exportConfig.streamStatements()) {
+            long timeout = exportConfig.getTimeoutSeconds();
```

Comment:
Can we reuse the streaming exporter code across all the export formats (CSV, JSON, GraphML) rather than having the same code multiple times?

I think the only thing that changes is the line that does the write/dump?

Reply:
Done!